play/src/components/GameSessionSwitch.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from 'react';
+import {
+  ApiClient,
+  IChoice,
+  IQuestion,
+  IGameSession,
+  GameSessionState,
+} from '@righton/networking';
+import { v4 as uuidv4 } from 'uuid';
+import PregameCountdown from '../pages/PregameCountdown';
+import GameInProgress from '../pages/GameInProgress';
+import PhaseResults from '../pages/PhaseResults';
+import FinalResultsContainer from '../containers/FinalResultsContainer';
+import StartPhase2 from '../pages/StartPhase2';
+import { PregameModel } from '../lib/PlayModels';
+
+interface GameInProgressContainerProps {
+  apiClient: ApiClient;
+  gameSession: IGameSession;
+  pregameModel: PregameModel;
+}
+
+export default function GameInProgressContainer({
+  apiClient,
+  gameSession,
+  pregameModel,
+}: GameInProgressContainerProps) {
+  const [isPregameCountdown, setIsPregameCountdown] = useState<boolean>(true);
+  const { currentState } = gameSession;
+  const currentQuestion =
+    gameSession.questions[gameSession.currentQuestionIndex ?? 0];
+  const currentTeam = gameSession.teams?.find(
+    (team) => team.id === pregameModel.teamId
+  );
```

Comment:
Totally agree with this. How do you mean force unwrap these? I'm doing these kind of messy checks because the IGameSession interface has an optional flag on currentQuestionIndex and teams. From a `play` perspective, this isn't relevant (as they are both always going to be present) but from a `host` and `central` perspective they can be `null`. 

Is there a better way of handling this condition?

Reply:
Sorry I'm using Swift terminologies here, not sure what it's called in TS, using `!` instead of `?` in here. So generally, we want to have only one source to check for nullability or enforce it here. I'm also thinking if it makes sense to have a different models or annotations here and handle these in networking, like if it's host app, it returns nullable and if it's play, it returns non-nullable.

We might be able to add helper methods to `IGameSession` as well something like:

```typescript
get nonNullTeams(): Array<ITeam> {
    if (isNullOrUndefined(teams) {
        console.error("Teams can't be null")
        return []
    }
    return teams // Typescript should be smart enough to know this is neither `null` nor `undefined`
}
```

Reply:
Yeah definitely, I think we should revisit on what are the next pain points are after separating containers and how we can help with reducing boilerplate and unnecessary null checks